src/main/java/io/confluent/idesidecar/restapi/credentials/ScramCredentials.java

```diff
@@ -0,0 +1,121 @@
+package io.confluent.idesidecar.restapi.credentials;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import io.confluent.idesidecar.restapi.exceptions.Failure;
+import io.confluent.idesidecar.restapi.exceptions.Failure.Error;
+import io.soabase.recordbuilder.core.RecordBuilder;
+import jakarta.validation.constraints.NotNull;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+@Schema(description = "Scram authentication credentials")
+@RecordBuilder
+public record ScramCredentials(
+    @Schema(description = "Hash algorithm")
+    @JsonProperty(value = "hash_algorithm")
+    @NotNull
+    HashAlgorithm hashAlgorithm,
+
+    @JsonProperty(value="scram_username")
+    @NotNull
+    String username,
+    @JsonProperty(value="scram_password")
+    @NotNull
+    Password password
+) implements Credentials {
+
+  public enum HashAlgorithm {
+    SCRAM_SHA_256("SCRAM-SHA-256"),
+    SCRAM_SHA_512("SCRAM-SHA-512");
+
+    private final String value;
+
+    HashAlgorithm(String value) {
+      this.value = value;
+    }
+
+    public String getValue() {
+      return value;
+    }
+  }
+
+  @Override
+  public Type type() {
+    return Type.SCRAM;
+  }
+
+  private static final String SCRAM_LOGIN_MODULE_CLASS = "org.apache.kafka.common.security.scram.ScramLoginModule";
+
+  @Override
+  public Optional<Map<String, String>> kafkaClientProperties(
+      KafkaConnectionOptions options
+  ) {
+
+    var jaasConfig = "%s required username=\"%s\" password=\"%s\";".formatted(
+        SCRAM_LOGIN_MODULE_CLASS,
+        username,
+        password.asString(options.redact())
+    );
+
+    var config = new LinkedHashMap<String, String>();
+    config.put("sasl.jaas.config", jaasConfig);
+    config.put("sasl.mechanism", hashAlgorithm.getValue());
```

Comment:
We could also use a constant for this config:

```suggestion
    config.put(SaslConfigs.SASL_MECHANISM, hashAlgorithm.getValue());
```